### PR TITLE
Test one off flow with placeholders and clean up code

### PR DESCRIPTION
**Pull request by CrystalPea**
### Work done:
- Start testing creating, editing and deleting an sms template
- Start testing one-off and test flow for email template with multiple placeholders including email address as a placeholder
- Start testing one-off and test flow for sms template with multiple placeholders including phone number as a placeholder
- Refactor the code, move some non-reusable tests out of utils and write reusable components for creating sms and email template, for deleting template and for sending a one-off message.

### CODE REVIEW

File: tests/functional/preview_and_dev/test_seeded_user.py [hunk 1]
```diff
@@ -78,99 +80,112 @@ def test_send_csv(driver, login_seeded_user, seeded_client, seeded_client_using_
...
+    placeholders = send_notification_to_one_recipient(
+        driver, template_name, "sms", test=False, recipient_data='07700900998', placeholders_number=2
+    )
+    assert list(placeholders[0].keys()) == ["name"]
+    assert list(placeholders[1].keys()) == ["things"]
+    delete_template(driver, template_name)
+
+
 @recordtime
 def test_send_sms_to_one_recipient(driver, login_seeded_user):
```
Comment (leohemsted) [inline tests/functional/preview_and_dev/test_seeded_user.py]: do we still need this test? seems like it largely duplicates the new test_send_sms_with_placeholders_to_one_recipient, can we just combine the asserts?

(the same with the email tests of the same name)

Comment (CrystalPea) [reply tests/functional/preview_and_dev/test_seeded_user.py]: You are right, I should be more bold :D

Comment (CrystalPea) [reply tests/functional/preview_and_dev/test_seeded_user.py]: Done in https://github.com/alphagov/notifications-functional-tests/pull/253/commits/b29ecaea3a8a8c576b430e1a8b2ac59ce96b7724 ☺️
